examples/tls_boring_dynamic_certs.rs

```diff
@@ -150,17 +148,14 @@ impl DynamicCertIssuer for DynamicIssuer {
         _server_name: Option<Host>,
     ) -> Result<ServerAuthData, OpaqueError> {
         match client_hello.ext_server_name() {
-            Some(host) => match host {
-                rama::net::address::Host::Name(domain) => {
-                    if domain == &Domain::from_static("example") {
-                        return Ok(self.example_data.clone());
-                    } else if domain == &Domain::from_static("second.example") {
-                        return Ok(self.second_example_data.clone());
-                    }
-                    Ok(self.example_data.clone())
+            Some(domain) => {
+                if domain == &Domain::from_static("example") {
```

Comment:
https://ramaproxy.org/docs/rama/net/address/struct.Domain.html#impl-PartialEq%3C%26str%3E-for-Domain is implemented so you can just do `domain == "example"` I know it's not your fault but you might as well improve this code while you're at it